core/controllers/editor.py

```diff
@@ -397,29 +387,54 @@ def delete(self, exploration_id):
 class ExplorationRightsHandler(EditorHandler):
     """Handles management of exploration editing rights."""
 
-    @require_editor
+    @acl_decorators.can_publish_exploration
```

Comment:
Why are we putting decorators on internal methods? Can we avoid doing this? You might need a less granular permission, or, if that doesn't make sense, to refactor this into a few separate handlers instead (which is totally fine). Make sure to modify the frontend accordingly and also test the end-to-end flow.